src/is-where.js

```diff
@@ -1,28 +1,72 @@
 const entries = require('object.entries');
 
+const ATTRIBUTE_PRESENT = {exists: true};
+
 const _isWhere = (where, target) => {
+  // Check each key from where
   for (const [key, value] of entries(where)) {
+
+    // If the key is set, but value is undefined, we ignore it
+    if (typeof value === 'undefined') {
+      continue;
+    }
+
+    // Allow a way to check for the pressenece of an attribute with that name.
+    if (value === ATTRIBUTE_PRESENT) {
+      if (!(key in target)) {
+        return false;
+      }
+      continue;
+    }
+
+    // Array checks
+    if (Array.isArray(value)) {
+      // [x,y,z] - allow the target value to be x, y, OR z
+      if (value.length && !value.some(test => target[key] === test)) {
+        return false;
+      }
+      continue;
+    }
+
+    // Null check
+    if (value === null) {
+      if (target[key] !== null) {
+        return false;
+      }
+      continue;
+    }
+
+    // Object checks (recursion)
     if (typeof value === 'object') {
```

Comment:
@gnarf wouldn't it be better to make this `if` an `else if` to the previous `if (value === null) {}`.
I think that `!(Boolean(target[key])` takes care of excluding `null` value, but it is less clear than making that:
```
if (value === null) {
}
else if (typeof value === 'object') {
  if (_isWhere(value, target[key])) {
  }
}
```

Reply:
All the previous `if` now `continue` so the `else` became meaningless worth it.

If any of the previous `if` caught them, they either `return false` for failing the check or `continue` for passing.

Reply:
Careful here also, the `Boolean(target[key])` is on the target, not the value - so if `value` is an object, and the target key is falsey, we bail, but otherwise we recurse into another `_isWhere` call